src/minimega/vnc_cli.go

```diff
@@ -27,72 +28,106 @@ If playback is selected, the specified file (created using vnc record) will be
+			"vnc <kb,fb> <record,> <vm name> <filename>",
+			"vnc <kb,fb> <norecord,> <vm name>",
+			"vnc <playback,> <vm name> <filename>",
+			"vnc <noplayback,> <vm name>",
 		},
-		Call: wrapSimpleCLI(cliVNC),
+		Call: wrapVMTargetCLI(cliVNC),
 	},
-	{ // clear vnc
+	{
 		HelpShort: "reset VNC state",
 		HelpLong: `
 Resets the state for VNC recordings. See "help vnc" for more information.`,
 		Patterns: []string{
 			"clear vnc",
 		},
-		Call: wrapSimpleCLI(func(_ *minicli.Command, _ *minicli.Response) error {
-			return vncClear()
-		}),
+		Call: wrapBroadcastCLI(cliVNCClear),
+	},
+	{
+		HelpShort: "list all running vnc playback/recording instances",
+		HelpLong: `
+List all running vnc playback/recording instances. See "help vnc" for more information.`,
+		Patterns: []string{
+			"vnc",
+		},
+		Call: wrapBroadcastCLI(cliVNCList),
 	},
 }
 
+// List all active recordings and playbacks
+func cliVNCList(c *minicli.Command, resp *minicli.Response) error {
+	resp.Header = []string{"host", "name", "id", "type", "filename"}
+	resp.Tabular = [][]string{}
+
+	for _, v := range vncKBRecording {
+		if inNamespace(v.Vm) {
+			resp.Tabular = append(resp.Tabular, []string{
+				hostname, v.Vm.Name, strconv.Itoa(v.Vm.ID),
+				"record kb",
+				v.file.Name(),
+			})
+		}
+	}
+
+	for _, v := range vncFBRecording {
+		if inNamespace(v.Vm) {
+			resp.Tabular = append(resp.Tabular, []string{
+				hostname, v.Vm.Name, strconv.Itoa(v.Vm.ID),
+				"record fb",
+				v.file.Name(),
+			})
+		}
+	}
+
+	for _, v := range vncKBPlaying {
+		if inNamespace(v.Vm) {
+			resp.Tabular = append(resp.Tabular, []string{
+				hostname, v.Vm.Name, strconv.Itoa(v.Vm.ID),
+				"playback kb",
+				v.file.Name(),
+			})
+		}
+	}
+
+	return nil
+}
+
 func cliVNC(c *minicli.Command, resp *minicli.Response) error {
-	host := c.StringArgs["host"]
-	vm := c.StringArgs["vm"]
 	fname := c.StringArgs["filename"]
 
-	if host == Localhost {
-		host = hostname
+	vm, err := vms.FindKvmVM(c.StringArgs["vm"])
+	if err != nil {
+		return fmt.Errorf("vm %s not found", c.StringArgs["vm"])
 	}
 
 	if c.BoolArgs["record"] && c.BoolArgs["kb"] {
 		// Starting keyboard recording
-		return vncRecordKB(host, vm, fname)
+		return vncRecordKB(vm.Name, fname)
 	} else if c.BoolArgs["record"] && c.BoolArgs["fb"] {
 		// Starting framebuffer recording
-		return vncRecordFB(host, vm, fname)
+		return vncRecordFB(vm.Name, fname)
 	} else if c.BoolArgs["norecord"] || c.BoolArgs["noplayback"] {
-		var err error
 		var client *vncClient
+		ns := fmt.Sprintf("%v:%v", vm.Namespace, vm.Name)
 
 		if c.BoolArgs["norecord"] && c.BoolArgs["kb"] {
-			err = fmt.Errorf("kb recording %v %v not found", host, vm)
-			for k, v := range vncKBRecording {
-				if v.Matches(host, vm) {
-					client = v.vncClient
-					delete(vncKBRecording, k)
-					break
-				}
+			err = fmt.Errorf("kb recording %v not found", vm.Name)
+			if v, ok := vncKBRecording[ns]; ok {
+				client = v.vncClient
+				delete(vncKBRecording, ns)
```

Comment:
Is this the right place to delete from `vncKBRecording`? Need to think about this.